File: Aug13/Troy/Q3.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        for(int row = 1; row <= numRows; row++ ){
            List<Integer> temp = new ArrayList<Integer>();
            for(int col = 1; col <= row; col++ ){
                if (row <= 2) {
                    temp.add(1);
                } else {
                    if (col == 1 || col == row) {
                        temp.add(1);
                    } else {
                        temp.add(result.get(row - 1 - 1).get(col - 1 - 1) + result.get(row - 1 - 1).get(col - 1));
                    }
                }
            }
            result.add(temp);
        }
        return result;
    }
}

File: 2018/Aug13/Carl/Q3.py
class Solution(object):
    def generate(self, numRows):  # 3
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        if numRows <= 0:
            return []
        pascal = [[1]]
        row = []
        if numRows == 1:
            return pascal
        for rowNum in range(1, numRows):  # 1, 2
            row.append(1)
            for rowIndex in range(1, rowNum):  # 1
                row.append(pascal[rowNum-1][rowIndex-1] +
                           pascal[rowNum-1][rowIndex])
            row.append(1)
            pascal.append(row)
            row = []
        return pascal


s = Solution()
print(s.generate(-1))
